llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -132,12 +134,60 @@ static bool expandRcpIntrinsic(CallInst *Orig) {
   return true;
 }
 
+static Intrinsic::ID getCorrectMaxIntrinsic(Type *elemTy,
+                                            Intrinsic::ID clampIntrinsic) {
```

Comment:
I could see us needing to get a max intrinsic in some other context at some point. Maybe a more specific name like `getMaxForClamp` is better.

Reply:
Will do.

As an aside this function started off generic until I found out I couldn't use `getSignBit()` to distinguish unsigned clamp vs signed clamp. The only way I could determine signed-ness was in codeGen.